File: app/src/main/java/com/example/applecustomer/fusedlocationprovider/LocationService.java
package com.example.applecustomer.fusedlocationprovider;

import android.app.IntentService;
import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.location.Location;
import android.location.LocationManager;
import android.os.Bundle;
import android.os.IBinder;
import android.support.annotation.Nullable;
import android.util.Log;

import com.google.android.gms.location.LocationResult;

import java.util.List;

public class LocationService extends IntentService

{
    public static final String ACTION_PROCESS_UPDATES = "123";

    public LocationService(String name) {
        super(name);
    }

    @Override
    protected void onHandleIntent(@Nullable Intent intent) {
        if (intent != null) {
            final String action = intent.getAction();
            if (ACTION_PROCESS_UPDATES.equals(action)) {
                LocationResult result = LocationResult.extractResult(intent);
                if (result != null) {
                    Location location = result.getLastLocation();
                       String longitude = String.valueOf(location.getLongitude());
                       String latitude = String.valueOf(location.getLatitude());

                    SharedPreferences.Editor editor = getSharedPreferences("location",MODE_PRIVATE).edit();
                    editor.putString("lat", longitude);
                    editor.putString("lon",latitude);
                    editor.apply();
                }
            }
        }
    }

}

File: app/src/main/java/com/example/applecustomer/fusedlocationprovider/LocationService.kt
package com.example.applecustomer.fusedlocationprovider

import android.app.IntentService
import android.app.NotificationManager
import android.app.Service
import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.location.Location
import android.location.LocationManager
import android.media.RingtoneManager
import android.net.Uri
import android.os.Bundle
import android.os.IBinder
import android.support.v4.app.NotificationCompat
import android.util.Log

import com.google.android.gms.location.LocationResult

class LocationService(name: String) : IntentService(name) {

    override fun onHandleIntent(intent: Intent?) {
        if (intent != null) {
            Log.d("Service", "In service")

            val action = intent.action
            if (ACTION_PROCESS_UPDATES == action) {


                    val longitude = intent.getStringExtra("lat")
                    val latitude =intent.getStringExtra("lon")

                    val editor = getSharedPreferences("location", Context.MODE_PRIVATE).edit()
                    editor.putString("lat", longitude)
                    editor.putString("lon", latitude)
                    editor.apply()
                    //   showNotification(longitude,latitude);
                }

        }
    }

    companion object {
        const val ACTION_PROCESS_UPDATES = "123"
    }


}

